### ISSUE TITLE
Refactor ireps_generator.py and add a high level API for generating ada code

### ISSUE THREAD

**Issue by hannes-steffenhagen-diffblue** [diffblue/gnat2goto#56]
Right now `ireps_generator` is a 2k+ LOC file with lots of state and write block interspersed everywhere. For maintainability it'd be better to split it up into several files by tasks.

Also, the way we generate the code right now we just have a bunch of write statements everywhere which are a bit hard to follow, I'm thinking if we had something like this (pseudocode):

```
irep_id_arg = ada.variable(name = 'S', type = ada.string_type)
trivial_irep_function = ada.function(name = 'Trivial_Irep', args = [irep_id_arg])
trivial_irep_result = ada.variable(name = 'V', type = json_value_type)
trivial_irep_function.declare(trivial_irep_result)
trivial_irep_function.body([
  v.method_call('Set_Field', ['id', irep_id_arg]),
  v.method_call('Set_Field', ['sub', empty_array.call()]),
  v.method_call('Set_Field', ['namedSub', create_object.call()]),
  v.method_call('Set_Field', ['comment', create_object.call()]),
  ada.return(v)
])
ireps_module.define_private_function(trivial_irep_function)
```

to produce code like this:

```
   function Trivial_Irep (S : String) return JSON_Value is
   begin
      return V : constant JSON_Value := Create_Object do
         V.Set_Field ("id", S);
         V.Set_Field ("sub", Empty_Array);
         V.Set_Field ("namedSub", Create_Object);
         V.Set_Field ("comment", Create_Object);
      end return;
   end Trivial_Irep;
```

Instead of having a bunch of `file.write("Some_Ada_Code")` calls everywhere.

Basically, have a python side Ada AST and print it. I'm not suggesting we do this _now_, but that'll probably be easier to maintain than having random print calls throughout the code everywhere so I'd prefer if we moved to something like that if we find the time to do it.
